### PR TITLE
Update 2.5D GDScript demo to display the editor plugin in Godot 4.

### PR DESCRIPTION
**Pull request by chrisl8**
This works now in Godot 4, but I still get this error when it starts:

```
  .\scene/main/node.h:445 - Parameter "data.tree" is null.
  res://addons/node25d/main_screen/viewport_25d.gd:23 - Invalid access to property or key 'process_frame' on a base object of type 'null instance'.
```

I'm not sure how to fix it though. I feel like there is something hacky about this line, but I'm not sure what the correct solution is:

https://github.com/godotengine/godot-demo-projects/blob/d190e73ad3af67447bb43ebe1ae0bcd01152e149/misc/2.5d/addons/node25d/main_screen/viewport_25d.gd#L20-L22

This should fix:
https://github.com/godotengine/godot-demo-projects/issues/1040
and
https://github.com/godotengine/godot-demo-projects/issues/1016

I did hit the files I edited with the [gdformat](https://github.com/Scony/godot-gdscript-toolkit) as well. If you prefer the changes without white-space updates I can remove those changes.

UPDATE: The error mentioned above **only** seems to happen if you open the `main_screen_25d` scene in the editor, then leave it open, and close and open the editor again.
image

### CODE REVIEW

File: misc/2.5d/addons/node25d/main_screen/gizmo_25d.gd [hunk 1]
```diff
@@ -70,7 +71,9 @@ func _process(_delta):
 		# Make sure the gizmo is located at the object.
 		global_position = node_25d.global_position
 		if ROUGHLY_ROUND_TO_PIXELS:
-			spatial_node.transform.origin = (spatial_node.transform.origin * Node25D.SCALE).round() / Node25D.SCALE
+			spatial_node.transform.origin = (
+				(spatial_node.transform.origin * Node25D.SCALE).round() / Node25D.SCALE
+			)
```
Comment (aaronfranke) [inline misc/2.5d/addons/node25d/main_screen/gizmo_25d.gd:76]: Would breaking the line up like this be sufficient to please the formatter?

```suggestion
			var pixel_pos: Vector3 = (spatial_node.transform.origin * Node25D.SCALE).round()
			spatial_node.transform.origin = pixel_pos / Node25D.SCALE
```

File: misc/2.5d/addons/node25d/main_screen/viewport_25d.gd [hunk 2]
```diff
@@ -123,14 +126,15 @@ func _gui_input(event):
 
 
 func _recursive_change_view_mode(current_node):
-	if current_node.has_method("set_view_mode"):
-		current_node.set_view_mode(view_mode_index)
-	for child in current_node.get_children():
-		_recursive_change_view_mode(child)
+	if current_node:
```
Comment (aaronfranke) [inline misc/2.5d/addons/node25d/main_screen/viewport_25d.gd:129]: Prefer using guard clauses.

```suggestion
	if not current_node:
		return
```

File: misc/2.5d/addons/node25d/node25d_plugin.gd [hunk 3]
```diff
@@ -5,43 +5,53 @@ const MainPanel = preload("res://addons/node25d/main_screen/main_screen_25d.tscn
...
-	add_custom_type("ShadowMath25D", "CharacterBody3D", preload("shadow_math_25d.gd"), preload("icons/shadow_math_25d_icon.png"))
+	add_custom_type(
+		"YSort25D", "Node", preload("y_sort_25d.gd"), preload("icons/y_sort_25d_icon.png")
+	)
+	add_custom_type(
+		"ShadowMath25D",
+		"CharacterBody3D",
+		preload("shadow_math_25d.gd"),
+		preload("icons/shadow_math_25d_icon.png")
+	)
```
Comment (aaronfranke) [inline misc/2.5d/addons/node25d/node25d_plugin.gd:28]: IMO this part hurts readability, I would undo the formatter's changes here.

File: misc/2.5d/addons/node25d/main_screen/gizmo_25d.gd (hunk 1 above)
Comment (chrisl8) [reply misc/2.5d/addons/node25d/main_screen/gizmo_25d.gd]: I just undid the formatting for these lines to put it back the way it was, if that is OK with you.

File: misc/2.5d/addons/node25d/node25d_plugin.gd (hunk 3 above)
Comment (chrisl8) [reply misc/2.5d/addons/node25d/node25d_plugin.gd]: I just undid the formatting for these lines to put it back the way it was, if that is OK with you.

File: misc/2.5d/addons/node25d/main_screen/viewport_25d.gd (hunk 2 above)
Comment (chrisl8) [reply misc/2.5d/addons/node25d/main_screen/viewport_25d.gd]: Done.

**Comment by chrisl8** [comment]
@aaronfranke  and @Calinou  Let me know if you need me to change anything else. Also if you need this squashed.

**Review by aaronfranke — APPROVED** [review]
Thanks for working on this! I pushed a few more changes to fully fix the shadow, fix the gizmo, and generally clean up the code.

**Comment by Calinou** [comment]
Note that I fixed incorrect static typing in a follow-up commit: https://github.com/godotengine/godot-demo-projects/commit/7223aec0012ba5f89b7f9f28a65e91abb30be0cc

**Pull request merged**